hcp-ui-templates/ecs/main.tf

```diff
@@ -0,0 +1,130 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 3.43"
+    }
+    hcp = {
+      source  = "hashicorp/hcp"
+      version = ">= 0.18.0"
+    }
+  }
+
+  provider_meta "hcp" {
+    module_name = "hcp-consul"
+  }
+}
+
+provider "consul" {
+  address    = hcp_consul_cluster.main.consul_public_endpoint_url
+  datacenter = hcp_consul_cluster.main.datacenter
+  token      = hcp_consul_cluster_root_token.token.secret_id
+}
+
+locals {
+  vpc_region = "{{ .VPCRegion }}"
+  hvn_region = "{{ .HVNRegion }}"
+  cluster_id = "{{ .ClusterID }}"
+}
+
+provider "aws" {
+  region = local.vpc_region
+}
+
+data "aws_availability_zones" "available" {}
+
+module "vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+  version = "3.10.0"
+
+  name                 = "${local.cluster_id}-vpc"
+  cidr                 = "10.0.0.0/16"
+  azs                  = data.aws_availability_zones.available.names
+  private_subnets      = ["10.0.1.0/24", "10.0.2.0/24", "10.0.3.0/24"]
+  public_subnets       = ["10.0.4.0/24", "10.0.5.0/24", "10.0.6.0/24"]
+  enable_dns_hostnames = true
+  enable_nat_gateway   = true
+  single_nat_gateway   = true
+}
+
+resource "hcp_hvn" "main" {
+  hvn_id         = "${local.cluster_id}-hvn"
+  cloud_provider = "aws"
+  region         = local.hvn_region
+  cidr_block     = "172.25.32.0/20"
+}
+
+module "aws_hcp_consul" {
+  source = "hashicorp/hcp-consul/aws"
+
+  hvn             = hcp_hvn.main
+  vpc_id          = module.vpc.vpc_id
+  subnet_ids      = module.vpc.public_subnets
+  route_table_ids = module.vpc.public_route_table_ids
+}
+
+resource "hcp_consul_cluster" "main" {
+  cluster_id      = local.cluster_id
+  hvn_id          = hcp_hvn.main.hvn_id
+  public_endpoint = true
+  tier            = "development"
+}
+
+resource "consul_config_entry" "service_intentions" {
+  name = "*"
+  kind = "service-intentions"
+
+  config_json = jsonencode({
+    Sources = [
+      {
+        Action     = "allow"
+        Name       = "*"
+        Precedence = 9
+        Type       = "consul"
+      },
+    ]
+  })
+}
+
+resource "hcp_consul_cluster_root_token" "token" {
+  cluster_id = hcp_consul_cluster.main.id
+}
+
+module "aws_ecs_cluster" {
+  source  = "hashicorp/hcp-consul/aws//modules/hcp-ecs-client"
+  version = "~> 0.3.1"
```

Comment:
Are you thinking the next version would be `0.3.1`, or `0.4.0`? My understanding is this will only use the rightmost number to pick versions against, so this constraint would not find `0.4.0`. Not a huge deal either way.

Reply:
I was thinking that `0.3.1` is the next version, because adding a new example is not a breaking change. But happy to change it.

Reply:
So this is a breaking change because of the rename `subnet` -> `public_subnet`. `0.4.0` it is.